Update website configuration for GitHub Pages compatibility

KeithMoyer opened the pull request:
This pull request does multiple things:
 - Updates instructions to test using container image designed for testing GitHub Pages.
 - Removes the Gemfiles, as they're not used in GitHub Pages and cause issues when testing against environments that simulate it.
 - Cleans up some places where we were creating temporary files in the repository.

deeglaze commented on README.md:
```diff
@@ -1,46 +1,44 @@
-docker start asylo-jekyll -a -i
-# Clean up, only needed if you won't be previewing website changes for a long time or you want to start over:
-docker rm asylo-jekyll
+docker run -it --rm -p 4000:4000 -v ${PWD}:/srv/jekyll jekyll:jekyll:pages \
```
jekyll/jekyll:pages

KeithMoyer replied:
Done.

deeglaze commented on README.md:
````diff
@@ -1,46 +1,44 @@
 
 ```bash
-HTML-Proofer finished successfully.
+docker run -it --rm -v ${PWD}:/srv/jekyll jekyll:jekyll:pages rake test
````
jekyll/jekyll:pages

deeglaze replied:
This command simply hangs for me. I just see this
rm -fr _rakesite
mkdir _rakesite
bundle exec jekyll build --config _config.yml,_rake_config_override.yml

KeithMoyer replied:
Are you running it using the content from the pull request?  That output implies that you are not.

deeglaze commented on README.md:
```diff
@@ -1,46 +1,44 @@
-Alternatively, if you just want to develop locally w/o Docker/Kubernetes/Minikube, you can try installing Jekyll locally.
-You may need to install other prerequisites manually (which is where using the docker image shines). Here's an example of doing
-so for Mac OS X:
+Alternatively, if you just want to develop locally w/o
```
s#w/o#without#

KeithMoyer replied:
Done

deeglaze commented on README.md:
```diff
@@ -58,22 +56,25 @@ bundle exec jekyll serve
 
-You should run `scripts/linters.sh` prior to checking in your changes.
-This will run 3 tests:
+You should run `scripts/linters.sh` prior to checking in your changes. This will
```
This is the first I've seen linters.sh. That script should be documented in scripts' README.md with its prerequisites. How does one acquire mdspell, for example?

KeithMoyer replied:
No idea.  I didn't write this text; I only ran a markdown formatter on it. If we want to clean up or remove these part, let's do it separate from this unrelated pull request.

KeithMoyer committed:
Update README to use "pages" container

This site is served using GitHub Pages and the Jekyll environment used
to test the site should be a close simulation to what is present in
GitHub Pages. Fortunately, Jekyll provides a container specifically for
that purpose. Using it is considerably easier than using a standard
Jekyll container image.

KeithMoyer committed:
Remove Gemfiles

This site is intended to run in GitHub Pages, which does not use
Gemfiles. Further, having Gemfiles that are not in sync with the
Gems present on GitHub Pages interferes with using environments that
simulate GitHub Pages (e.g., jekyll/jekyll:pages Docker container).

Rather than trying to keep our Gemfiles continually updated to track
the Gem versions on GitHub Pages, it seems prudent to just remove them
since they don't provide much value in our use-case anyway.

KeithMoyer committed:
Remove errant baseurl from Jekyll configuration

The "baseurl" configuration is intended to include the portion of the
site URL after the domain and the Jekyll page path*. In the case where
you don't want anything in between, the default value of "" (not /) is
correct.

We could have "" in the configuration to explicitly reiterate the
default setting, but some of our website building machinery doesn't
handle quotations. Instead, it seems reasonable to just remove the
setting and implicitly use the default.

* https://byparker.com/blog/2014/clearing-up-confusion-around-baseurl/

KeithMoyer committed:
Remove HTMLProofer cache

The cache for HTMLProofer makes testing more efficient by only checking
for broken links that haven't been checked recently.

We don't have very many external links, and they can be checked very
quickly. So, this isn't buying much for us.

Removing this also has the benefit of no longer creating a cache
directory (tmp/.htmlproofer) in the repository. This was somewhat
problematic if using a docker container as it would be created owned by
root.

KeithMoyer committed:
Use separate temp directory in rake test

The existing rake test created a _rakesite directory in the repository
for building the site. Instead, create it under /tmp so it doesn't
interfere with the repository.

This is especially useful in the case of using a container to run the
rake tests, since the directory is created with root ownership.

KeithMoyer committed:
Format the README
